### PR TITLE
Add Feature to Gherkin topic

### PR DESCRIPTION
**Pull request by Javix**
* create gherkin.md
* add feature introduction section
* add step definitions section

### CODE REVIEW

**Comment by mattwynne** [comment]
Thanks @Javix - dd you write this from scratch or pull it from somewhere else? I have the feeling I've read some of this before.

**Comment by Javix** [comment]
Sure, I've taken it from the old wiki😊

File: docs/gherkin.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
+# Gherkin
+
+Gherkin is the language that Cucumber understands. It is a Business Readable, [Domain Specific Language](http://martinfowler.com/bliki/BusinessReadableDSL.html) that lets you describe software’s behaviour without detailing how that behaviour is implemented.
+
+Gherkin serves two purposes — documentation and automated tests. The third is a bonus feature — when it yells in red it’s talking to you, telling you what code you should write.
+
+Gherkin’s grammar is defined in the Treetop grammar that is part of the Cucumber codebase. The grammar exists in different flavours for many spoken languages (37 at the time of writing), so that your team can use the keywords in your own language.
```
Comment (jbpros) [inline docs/gherkin.md]: This is pretty technical for an introduction. I think all we should mention is that a lot of spoken languages (much more than 37 nowadays I believe) are supported. Keeping the part about why this matters (i.e. use the language of your team/project) is good though.

File: docs/gherkin.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+Like Python and YAML, Gherkin is a line-oriented language that uses indentation to define structure. Line endings terminate statements (eg, steps). Either spaces or tabs may be used for indentation (but spaces are more portable). Most lines start with a keyword.
+
+Comment lines are allowed anywhere in the file. They begin with zero or more spaces, followed by a hash sign (`#`) and some amount of text.
+
+The parser divides the input into features, scenarios and steps. When you run the feature the trailing portion (after the keyword) of each step is matched to a Ruby code block called [Step Definitions](docs/step-definitions.md).
+
+A Gherkin source file usually looks like this
+```
+1: Feature: Some terse yet descriptive text of what is desired
```
Comment (jbpros) [inline docs/gherkin.md]: Let's drop the line numbers, they're confusing. The explanations below can refer to the "line staring with Feature:", the text above it, etc.

File: docs/gherkin.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+Gherkin’s grammar is defined in the Treetop grammar that is part of the Cucumber codebase. The grammar exists in different flavours for many spoken languages (37 at the time of writing), so that your team can use the keywords in your own language.
+
+There are a few conventions.
+
+ * Single Gherkin source file contains a description of a single feature.
+ * Source files have `.feature` extension.
+
+## Gherkin Syntax
+
+Like Python and YAML, Gherkin is a line-oriented language that uses indentation to define structure. Line endings terminate statements (eg, steps). Either spaces or tabs may be used for indentation (but spaces are more portable). Most lines start with a keyword.
```
Comment (jbpros) [inline docs/gherkin.md]: Those three first paragraphs are a bit tough to digest, especially for beginners. I suggest we move those specificities down (or even drop them for now) and simply explain that .feature files are text files with a limited number of keywords and a lot of content specific to the user's domain: descriptions, business rules and examples?

File: docs/gherkin.md (hunk 1 above)
Comment (mattwynne) [inline docs/gherkin.md]: It's also out of date - we haven't used Treetop for several years now.

**Comment by mattwynne** [comment]
Not your fault @Javix but I think this content is pretty crap. It's over-technical and doesn't provide a simple, accessible introduction to Gherkin and how to use it.

I think we need to avoid lifting content as-is from the old wiki - we need to do some quality checking as we do.

File: docs/gherkin.md (hunk 1 above)
Comment (Javix) [inline docs/gherkin.md]: @mattwynne the `Treetop` mention has been removed in #34.

**Comment by Javix** [comment]
@mattwynne sure, I'll try to do my best. I feel like an author preparing his book and who has to apply corrections provided by his editor, all that will always give the best result for the future readers :smile:

**Comment by mattwynne** [comment]
I know it must be tiring @Javix and thanks for your patience. I think we'll soon build some momentum and things will feel more satisfying for you.

**Comment by Javix** [comment]
@mattwynne no worries, this is my real first experience of writing docs, especially for the tool I love. So I'm patient and shall `absorb` everything to enlarge my Xperience :smile:

**Comment by aslakhellesoy** [comment]
This is obsolete - superseded by #34

**Pull request closed without merging**